prototypes/toggle-controls/generate-reader.py

```diff
@@ -3,91 +3,60 @@
 import os
 import shutil
 
-from pysblgnt import morphgnt_rows
+from jinja2 import Environment, FileSystemLoader
 
+from utils import rows_by_verses_by_chapters_for_book
 
-def generate(book_title, book_num, output_filename):
+
+env = Environment(
+    loader=FileSystemLoader("."),
+)
+template = env.get_template("template.html")
+
+
+def before(row):
+    word = row["word"]
+    text = row["text"]
+    return text[:text.index(word)]
+
+
+def after(row):
```

Comment:
Oh wow, that's neat. Never occurred to me that would work but it's perfect. I wonder if Jinja's `set` supports tuple assigment. e.g. `{% set before, after = text.split(word, 1) %}`. I suspect not but worth trying. Of course, doing `before = text.split(word, 1)[0]` and `after = text.split(word, 1)[1]` separately might still be better than what we have.

Although I'm still looking for alternatives to passing in functions to the context (and somewhat oscillate between doing stuff in-line in Jinja versus in separate Python files)